### ISSUE TITLE
Implement startPosition prop to set the start time of a video

### ISSUE THREAD

**Issue by jayfeldman12** [TheWidlarzGroup/react-native-video#2778]
# Feature Request

Add a new prop for iOS and Android to be able to set the start time a video opens at. I am open to doing the PR for this, but per the contribution guide I'm opening an issue first.

## Why it is needed

Sometimes users may want to have a video start at a time other than 0:00. It's possible to manually run a .seek in the onLoad prop, but on Android in particular for bigger video files, there is noticeable lag in this starting.  

## Possible implementation

In my local project we used patch-package to make this work on android. In ReactExoplayerView.java in initializePlayerSource, the start time can be used if resumePosition is not present.

Logic to set the start time in getVideoTrackInfoFromManifest would likely be needed to initialize start time as well, something like 
final long startTime = (this.startTime > 0 ? this.startTime : this.contentStartTime) * 1000 - 100; // s -> ms with 100ms offset
But I don't understand that section quite as well.

I haven't looked into the swift side yet and don't have as much knowledge there, but I believe in the setter that would be created, you could set pendingSeek and pendingSeekTime if startTime is passed in, then handleReadyToPlay would automatically seek to the correct time when the player is ready.

### Code sample

After setting the startTime variable, snippet from our android patch-package:
```
@@ -624,11 +625,14 @@ class ReactExoplayerView extends FrameLayout implements
             }
         }
 
-        boolean haveResumePosition = resumeWindow != C.INDEX_UNSET;
-        if (haveResumePosition) {
+        boolean useStartTime = this.startTime > 0;
+        boolean useResumePosition = resumeWindow != C.INDEX_UNSET && resumePosition > 0;
+        if (useResumePosition) {
             player.seekTo(resumeWindow, resumePosition);
+        } else if (useStartTime) {
+            player.seekTo(this.startTime * 1000); // sec to ms
         }
-        player.prepare(mediaSource, !haveResumePosition, false);
+        player.prepare(mediaSource, !useResumePosition && !useStartTime, false);
         playerNeedsSource = false;
```

For iOS, I haven't done much native work but something like this may work

@objc
  func setStartTime(_ startTime:Float) {
      if startTime > 0 {
          _pendingSeek = true
          _pendingSeekTime = startTime
      }
  }

**Comment by hueniverse** [comment]
@freeboub What do you think?

**Comment by freeboub** [comment]
I think this is already implemented on android and this feature is welcome on ios (contentStartTime):
[contentStartTime](https://github.com/react-native-video/react-native-video/blob/master/API.md#contentStartTime)

Let's tag this issue as feature request.

**Comment by jayfeldman12** [comment]
> I think this is already implemented on android and this feature is welcome on ios (contentStartTime): [contentStartTime](https://github.com/react-native-video/react-native-video/blob/master/API.md#contentStartTime)
> 
> Let's tag this issue as feature request.

So, based on the docs I thought that was meant to be something like what time the actual content of the video starts, and before that it was ads the video plays to set the resolution properly only at that time.

>The start time in ms for SSAI content. This determines at what time to load the video info like resolutions. Use this only when you have SSAI stream where ads resolution is not the same as content resolution. 

If I misunderstood and it is meant to set the time to start the video, then I believe it's not working on Android because when I used that prop, the video starts at the usual time.

**Comment by freeboub** [comment]
Heum, you are right.
I don't think this 'contentStartTime' is a good feature then ....
I can implement it next week for android, but I need help to support ios :/

**Comment by AlkanV** [comment]
@freeboub any update on this? i also need similar feature; ability to start a video from given position.

**Comment by imadarif** [comment]
contentStartTime is not working ? is there anyway i can set the video start time dynamically ? i have implemented slider so user can rewind and forward the video.

**Comment by dianamelga** [comment]
@freeboub do you have any update on this? facing the same issue while trying to start a video from a given position

**Comment by theycallmeaddy** [comment]
> @freeboub do you have any update on this? facing the same issue while trying to start a video from a given position

@dianamelga  Got any solution?

**Comment by freeboub** [comment]
I can confirm the proposed change works fine, but unfortunately I don't have time to backport it here.
There is maybe a small limitation linked to ghost image when launching the stream.

**Comment by freeboub** [comment]
Will beta available on 6.0.0-beta.1
